autocorpus/bioc_supplementary.py

```diff
@@ -0,0 +1,311 @@
+"""This module provides functionality for converting text extracted from various file types into a BioC format."""
+
+import datetime
+import re
+from dataclasses import dataclass, field
+from pathlib import Path
+from typing import Any
+
+import pandas as pd
+from pandas import DataFrame
+
+from .ac_bioc import (
+    BioCCollection,
+    BioCDocument,
+    BioCJSON,
+    BioCPassage,
+)
+from .ac_bioc.bioctable import (
+    BioCTableCell,
+    BioCTableCollection,
+    BioCTableDocument,
+    BioCTableJSON,
+    BioCTablePassage,
+)
+
+
+def extract_table_from_pdf_text(text: str) -> tuple[str, list[DataFrame]]:
```

Comment:
As this function is on the long side and essentially does two things, you could split it up into two subfunctions that you call from this one.